### PR TITLE
Disable diagnostic trust manager in tests

### PR DESCRIPTION
**Pull request by jkakavas**
This commit sets `xpack.security.ssl.diagnose.trust` to false in all
of our tests when running in FIPS 140 mode and when settings objects
are used to create an instance of the SSLService. This is needed
in 7.x because setting xpack.security.ssl.diagnose.trust to true
wraps SunJSSE TrustManager with our own DiagnosticTrustManager and
this is not allowed when SunJSSE is in FIPS mode.
An alternative would be to set xpack.security.fips.enabled to
true which would also implicitly disable
xpack.security.ssl.diagnose.trust but would have additional effects
(would require that we set PBKDF2 for password hashing algorithm in
all test clusters, would prohibit using JKS keystores in nodes even
if relevant tests have been muted in FIPS mode etc.)

Relates: #49900
Resoles: #51268

### CODE REVIEW

**Comment by jkakavas** [comment]
When we will start running tests in FIPS mode on Java 8 with both SunJSSE and BCJSSE, we can replace the `inFipsJvm()` checks with `inSunJSSEFipsJvm()` as the limitation applies only to SunJSSE

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/PkiRealmBootstrapCheckTests.java [hunk 1]
```diff
@@ -144,4 +145,12 @@ public void testBootstrapCheckWithClosedSecuredSetting() throws Exception {
         secureSettings.close();
         assertThat(check.check(createTestContext(settings, null)).isFailure(), Matchers.equalTo(expectFail));
     }
+
+    private Settings.Builder getSettingsBuilder(){
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/PkiRealmBootstrapCheckTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java [hunk 2]
```diff
@@ -98,10 +98,14 @@
         if (security != null) {
             throw new IllegalStateException("Security object already exists (" + security + ")");
         }
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder()
             .put("xpack.security.enabled", true)
             .put(testSettings)
-            .put("path.home", createTempDir()).build();
+            .put("path.home", createTempDir());
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 3]
```diff
@@ -1112,6 +1113,9 @@ public Client client(Node node, String clusterName) {
                 .put("logger.prefix", nodeSettings.get("logger.prefix", ""))
                 .put("logger.level", nodeSettings.get("logger.level", "INFO"))
                 .put(settings);
+            if (inFipsJvm()){
```
Comment (tvernum) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]:
```suggestion
            if (inFipsJvm()) {
```

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/TestXPackTransportClient.java [hunk 4]
```diff
@@ -51,4 +52,12 @@ public void close() {
             }
         }
     }
+
+    private static Settings possiblyDisableTlsDiagnostic(Settings settings){
```
Comment (tvernum) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/TestXPackTransportClient.java]:
```suggestion
    private static Settings possiblyDisableTlsDiagnostic(Settings settings) {
```

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/TestXPackTransportClient.java [hunk 5]
```diff
@@ -51,4 +52,12 @@ public void close() {
...
+        Settings.Builder builder = Settings.builder().put(settings);
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/TestXPackTransportClient.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecurityIntegTestCase.java [hunk 6]
```diff
@@ -246,6 +246,9 @@ protected Settings nodeSettings(int nodeOrdinal) {
         builder.put(LicenseService.SELF_GENERATED_LICENSE_TYPE.getKey(), "trial");
         builder.put(NetworkModule.TRANSPORT_TYPE_KEY, randomBoolean() ? SecurityField.NAME4 : SecurityField.NIO);
         builder.put(NetworkModule.HTTP_TYPE_KEY, randomBoolean() ? SecurityField.NAME4 : SecurityField.NIO);
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecurityIntegTestCase.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecuritySettingsSource.java [hunk 7]
```diff
@@ -147,6 +147,9 @@ public Settings nodeSettings(int nodeOrdinal) {
                 .put("xpack.security.authc.realms." + FileRealmSettings.TYPE + ".file.order", 0)
                 .put("xpack.security.authc.realms." + NativeRealmSettings.TYPE + ".index.order", "1")
                 .put("xpack.license.self_generated.type", "trial");
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecuritySettingsSource.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecuritySingleNodeTestCase.java [hunk 8]
```diff
@@ -167,6 +168,9 @@ protected Settings nodeSettings() {
         builder.put(LicenseService.SELF_GENERATED_LICENSE_TYPE.getKey(), "trial");
         builder.put("transport.type", "security4");
         builder.put("path.home", customSecuritySettingsSource.nodePath(0));
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecuritySingleNodeTestCase.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/PkiRealmBootstrapCheckTests.java [hunk 9]
```diff
@@ -144,4 +145,12 @@ public void testBootstrapCheckWithClosedSecuredSetting() throws Exception {
         secureSettings.close();
         assertThat(check.check(createTestContext(settings, null)).isFailure(), Matchers.equalTo(expectFail));
     }
+
+    private Settings.Builder getSettingsBuilder(){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/PkiRealmBootstrapCheckTests.java]:
```suggestion
    private Settings.Builder getSettingsBuilder() {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/oidc/TransportOpenIdConnectLogoutActionTests.java [hunk 10]
```diff
@@ -92,7 +92,11 @@ public void setup() throws Exception {
             .put(XPackSettings.TOKEN_SERVICE_ENABLED_SETTING.getKey(), true)
             .put("path.home", createTempDir())
             .build();
-        final Settings sslSettings = Settings.builder()
+        Settings.Builder sslSettingsBuilder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/oidc/TransportOpenIdConnectLogoutActionTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/CommandLineHttpClientTests.java [hunk 11]
```diff
@@ -70,7 +71,11 @@ public void testCommandLineHttpClientCanExecuteAndReturnCorrectResultUsingSSLSet
     }
 
     public void testGetDefaultURLFailsWithHelpfulMessage() {
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/CommandLineHttpClientTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/CommandLineHttpClientTests.java [hunk 12]
```diff
@@ -87,7 +92,11 @@ private MockWebServer createMockWebServer() {
     private Settings.Builder getHttpSslSettings() {
         MockSecureSettings secureSettings = new MockSecureSettings();
         secureSettings.setString("xpack.security.http.ssl.secure_key_passphrase", "testnode");
-        return Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/CommandLineHttpClientTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/ActiveDirectoryRealmTests.java [hunk 13]
```diff
@@ -142,7 +143,11 @@ public void start() throws Exception {
         }
         threadPool = new TestThreadPool("active directory realm tests");
         resourceWatcherService = new ResourceWatcherService(Settings.EMPTY, threadPool);
-        globalSettings = Settings.builder().put("path.home", createTempDir()).build();
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/ActiveDirectoryRealmTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapRealmTests.java [hunk 14]
```diff
@@ -98,7 +99,11 @@
     public void init() throws Exception {
         threadPool = new TestThreadPool("ldap realm tests");
         resourceWatcherService = new ResourceWatcherService(Settings.EMPTY, threadPool);
-        defaultGlobalSettings = Settings.builder().put("path.home", createTempDir()).build();
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapRealmTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapSessionFactoryTests.java [hunk 15]
```diff
@@ -59,7 +60,11 @@ public void setup() throws Exception {
         final Path origCa = getDataPath("/org/elasticsearch/xpack/security/authc/ldap/support/ldap-ca.crt");
         ldapCaPath = createTempFile();
         Files.copy(origCa, ldapCaPath, StandardCopyOption.REPLACE_EXISTING);
-        globalSettings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapSessionFactoryTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapUserSearchSessionFactoryTests.java [hunk 16]
```diff
@@ -62,8 +63,11 @@ public void init() throws Exception {
          * If we re-use an SSLContext, previously connected sessions can get re-established which breaks hostname
          * verification tests since a re-established connection does not perform hostname verification.
          */
-
-        globalSettings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapUserSearchSessionFactoryTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapTestCase.java [hunk 17]
```diff
@@ -172,6 +172,9 @@ public static Settings buildLdapSettings(RealmConfig.RealmIdentifier realmId, St
         if (serverSetType != null) {
             builder.put(getFullSettingKey(realmId, LdapLoadBalancingSettings.LOAD_BALANCE_TYPE_SETTING), serverSetType.toString());
         }
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapTestCase.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/SessionFactoryLoadBalancingTests.java [hunk 18]
```diff
@@ -292,7 +293,11 @@ private TestSessionFactory createSessionFactory(LdapLoadBalancing loadBalancing)
         Settings globalSettings = Settings.builder().put("path.home", createTempDir()).put(settings).build();
         RealmConfig config = new RealmConfig(REALM_IDENTIFIER, globalSettings,
                 TestEnvironment.newEnvironment(globalSettings), new ThreadContext(Settings.EMPTY));
-        return new TestSessionFactory(config, new SSLService(Settings.EMPTY, TestEnvironment.newEnvironment(config.settings())),
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/support/SessionFactoryLoadBalancingTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticatorTests.java [hunk 19]
```diff
@@ -93,7 +94,11 @@
 
     @Before
     public void setup() {
-        globalSettings = Settings.builder().put("path.home", createTempDir())
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticatorTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/AbstractSimpleSecurityTransportTestCase.java [hunk 20]
```diff
@@ -81,7 +81,11 @@ protected SSLService createSSLService(Settings settings) {
         secureSettings.setString("xpack.security.transport.ssl.secure_key_passphrase", "testnode");
         // Some tests use a client profile. Put the passphrase in the secure settings for the profile (secure settings cannot be set twice)
         secureSettings.setString("transport.profiles.client.xpack.security.ssl.secure_key_passphrase", "testnode");
-        Settings settings1 = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/AbstractSimpleSecurityTransportTestCase.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterIntegrationTests.java [hunk 21]
```diff
@@ -205,4 +205,12 @@ public String executor() {
         }
     }
 
+    private Settings.Builder getSettingsBuilder(){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterIntegrationTests.java]:
```suggestion
    private Settings.Builder getSettingsBuilder() {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterIntegrationTests.java [hunk 22]
```diff
@@ -205,4 +205,12 @@ public String executor() {
...
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterIntegrationTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/filter/IPFilterTests.java [hunk 23]
```diff
@@ -265,7 +266,11 @@ public void testThatAllAddressesAreAllowedWhenLicenseDisablesSecurity() {
     }
 
     public void testThatNodeStartsWithIPFilterDisabled() throws Exception {
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/filter/IPFilterTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java [hunk 24]
```diff
@@ -47,7 +47,11 @@ public void createSSLService() {
 
         MockSecureSettings secureSettings = new MockSecureSettings();
         secureSettings.setString("xpack.security.http.ssl.secure_key_passphrase", "testnode");
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java [hunk 25]
```diff
@@ -150,7 +154,11 @@ public void testCustomSSLConfiguration() throws Exception {
     public void testNoExceptionWhenConfiguredWithoutSslKeySSLDisabled() throws Exception {
         MockSecureSettings secureSettings = new MockSecureSettings();
         secureSettings.setString("xpack.security.http.ssl.secure_key_passphrase", "testnode");
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SecurityNioHttpServerTransportTests.java [hunk 26]
```diff
@@ -55,7 +55,11 @@ public void createSSLService() {
         Path testNodeCert = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.crt");
         MockSecureSettings secureSettings = new MockSecureSettings();
         secureSettings.setString("xpack.security.http.ssl.secure_key_passphrase", "testnode");
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SecurityNioHttpServerTransportTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SecurityNioHttpServerTransportTests.java [hunk 27]
```diff
@@ -183,7 +187,11 @@ public void testCustomSSLConfiguration() throws IOException {
     public void testNoExceptionWhenConfiguredWithoutSslKeySSLDisabled() {
         MockSecureSettings secureSettings = new MockSecureSettings();
         secureSettings.setString("xpack.security.http.ssl.truststore.secure_password", "testnode");
-        Settings settings = Settings.builder()
+        Settings.Builder builder = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SecurityNioHttpServerTransportTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ssl/SslIntegrationTests.java [hunk 28]
```diff
@@ -118,6 +118,9 @@ public void testThatTransportClientUsingSSLv3ProtocolIsRejected() {
 
     public void testThatConnectionToHTTPWorks() throws Exception {
         Settings.Builder builder = Settings.builder().put("xpack.security.http.ssl.enabled", true);
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ssl/SslIntegrationTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageCertificateVerificationTests.java [hunk 29]
```diff
@@ -184,6 +186,9 @@ private void enableHttpsHostnameChecking(SSLSocket clientSocket) {
             .put(prefix + ".key", getPath(keyPath))
             .put(prefix + ".client_authentication", clientAuth.name())
             .put(prefix + ".verification_mode", verificationMode.name());
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageCertificateVerificationTests.java]:
```suggestion
        if (inFipsJvm()) {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java [hunk 30]
```diff
@@ -378,4 +379,12 @@ private String randomSslPrefix() {
             "xpack.monitoring.exporters.http.ssl"
         );
     }
+
+    private Settings.Builder getSettingsBuilder(){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java]:
```suggestion
    private Settings.Builder getSettingsBuilder() {
```

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java [hunk 31]
```diff
@@ -378,4 +379,12 @@ private String randomSslPrefix() {
...
+        final Settings.Builder settings = Settings.builder();
+        if (inFipsJvm()){
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageFileTests.java]:
```suggestion
        if (inFipsJvm()) {
```

**Review by tvernum — APPROVED** [review]
LGTM with 31 nits 😈

**Comment by jkakavas** [comment]
> LGTM with 31 nits 

That's what you get for depending too much on :robot: and then they become incompatible with your current IDE version.

**Pull request merged**